### DISCUSSION TITLE
New renovate-ce image uid/guid settings

### DISCUSSION THREAD

**Discussion by meenzen** [mend/renovate-ce-ee#339] [Q&A] [answered]
According to the announcement email this new version has `100% backwards compatibility`. This does not appear to be the case, I have found these issues:

### Broken Docker Volumes

The new image does not use the same `uid` and `gid` as before, thus breaking the volume mounts

**Comment by rarkins** [comment, marked as answer]
Thanks, please keep separate topics in separate issues so that we can deal with them one by one and not miss any.

Starting with the UID/GID topic, I see:

```
❯ docker run --rm -it whitesource/renovate:5.0.0 /bin/bash
ubuntu@d5b138a78ef3:/usr/src/app$ id
uid=1000(ubuntu) gid=0(root) groups=0(root),1000(ubuntu)
```

and

```
❯ docker run --rm -it ghcr.io/mend/renovate-ce:6.0.0 /bin/bash
ubuntu@20221c835f88:/usr/src/app$ id
uid=1000(ubuntu) gid=0(root) groups=0(root),999(docker),1000(ubuntu)
```

I see the _addition_ of the `docker` group but wouldn't expect that to be breaking. What do you see/find?

**Comment by meenzen** [comment]
I may have jumped to the conclusion too early, sorry for the confusion.

I have configured a volume for `/db/`, exactly the same way as the volume for `/tmp/` and expected this to just work. Instead the app crashes:

```json
FATAL: failure initializing Renovate community edition - exiting...
       "err": {
         "errno": 14,
         "code": "SQLITE_CANTOPEN",
         "message": "SQLITE_CANTOPEN: unable to open database file",
         "stack": "Error: SQLITE_CANTOPEN: unable to open database file"
       }
```

The fix was to set the user explicitly, e.g. `user: "0:0"` so I just assumed that the user configuration of the image had changed.

Does SQLite require different file permissions than the `/tmp/` cache for some reason?

**Reply by rarkins** [reply]
I'm not really sure why that would be, but one of my colleagues can likely take a look next week. I think that user 0 means root so that's probably undesirable?

**Comment by rarkins** [comment]
Converted to discussion

**Comment by lluiscab** [comment]
Facing the same issue here, would like to get this fixed before upgrading to renovate CE

**Reply by rarkins** [reply]
The original description of this issue was _incorrect_. uid/gid is the same. 

What is "this"? e.g. mounting of DB? If so, please start a new discussion because this one is wrong/confusing

**Reply by lluiscab** [reply]
Facing the same issue as @meenzen 

> ```json
> FATAL: failure initializing Renovate community edition - exiting...
>        "err": {
>          "errno": 14,
>          "code": "SQLITE_CANTOPEN",
>          "message": "SQLITE_CANTOPEN: unable to open database file",
>          "stack": "Error: SQLITE_CANTOPEN: unable to open database file"
>        }
> ```

Setting  the user explicitly to `user: "0:0"` as @meenzen also said, fixes the issue. Should I open a new issue or do you want to edit the title / description of the current one?

**Reply by rarkins** [reply]
Please start a new discussion, including all the relevant info. Do not link to this confused discussion. I will mark this particular one as resolved.
